DataDog/datadog-agent
# logs/sender: add telemetry tagged by host & error

### What does this PR do?

Add new `logs_client_http_destination` metric to track HTTP payloads submitted by host and error type.

### Motivation

This will help troubleshoot whether payloads are being sent/retried/dropped for each destination. 

This is a minor follow-up to https://github.com/DataDog/datadog-agent/pull/7735. 

### Describe your test plan

1. Run an agent build with https://github.com/DataDog/integrations-core/pull/9045 included
    1. Configure the postgres or mysql check to collect statement samples as described in https://github.com/DataDog/integrations-core/pull/8627 
    1. Enable agent telemetry `telemetry.enabled: true`
1. Validate telemetry: `curl http://localhost:5000/telemetry | grep logs_client_http_destination` to confirm that the `logs_client_http_destination` metrics appear with `endpoint-host` and `error` tags.
